src/visualization/src/FrameVisualization.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2021 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+#ifndef IDYNTREE_FRAMEVISUALIZATION_H
+#define IDYNTREE_FRAMEVISUALIZATION_H
+
+#include <iDynTree/Visualizer.h>
+
+#include <vector>
+#include <irrlicht.h>
+
+namespace iDynTree
+{
+    class FrameVisualization : public IFrameVisualization
+    {
+        struct Frame
+        {
+            irr::scene::ISceneNode * visualizationNode = nullptr;
+        };
+
+        std::vector<Frame> m_frames;
+        irr::scene::ISceneManager* m_smgr;
+
+        void setFrameTransform(size_t index, const Transform& transformation);
+
+    public:
+
+        FrameVisualization();
+
+        ~FrameVisualization();
+
+        void init(irr::scene::ISceneManager* smgr);
+
+        void close();
+
+        FrameVisualization(const FrameVisualization& other) = delete;
+
+        FrameVisualization& operator=(const FrameVisualization& other) = delete;
+
+        /**
+         * @brief Add a frame in the visualization
+         * @return The frame index.
+         */
+        virtual size_t addFrame(const Transform& transformation, double arrowLength = 1.0) final;
```

Comment:
Maybe it would also be convenient to have a method that removes a frame?

Reply:
I have tried adding the method to remove a frame, but the logic behind was not the best. Since I am returning an index as an identifier, I do not like what happens when you remove a frame. Should that index be reused or not? Also, with the ``getNumberOfFrames`` method you can understand whether an index is valid or not. But if you remove one frame, this does not hold anymore as the total number of frames will be lower or equal than the maximum index. Hence, for the time being, I preferred to allow controlling the visibility of a frame with  https://github.com/robotology/idyntree/pull/802/commits/98e804a2272868576bd5f3eee0ac51c39a547429. The net result is the same and the logic behind remains straightforward.

Reply:
Great!